rye/src/cli/rye.rs

```diff
@@ -52,12 +53,54 @@ pub struct Args {
     command: SubCommand,
 }
 
+#[derive(Clone, Debug, ValueEnum)]
+enum ShellCompletion {
+    /// Bourne Again SHell (bash)
+    Bash,
+    /// Elvish shell
+    Elvish,
+    /// Friendly Interactive SHell (fish)
+    Fish,
+    /// PowerShell
+    PowerShell,
+    /// Z SHell (zsh)
+    Zsh,
+    /// Nushell
+    Nushell,
+}
+
+impl Generator for ShellCompletion {
+    /// Generate the file name for the completion script.
+    fn file_name(&self, name: &str) -> String {
+        match self {
+            ShellCompletion::Nushell => Nushell.file_name(name),
+            ShellCompletion::Bash => Shell::Bash.file_name(name),
+            ShellCompletion::Elvish => Shell::Elvish.file_name(name),
+            ShellCompletion::Fish => Shell::Fish.file_name(name),
+            ShellCompletion::PowerShell => Shell::PowerShell.file_name(name),
+            ShellCompletion::Zsh => Shell::Zsh.file_name(name),
```

Comment:
I decided to just expand these rather than add an `unreachable!` but it's mostly personal preference.

Reply:
SGTM!